src/libcore/sync/atomic.rs

```diff
@@ -1575,6 +1575,41 @@ assert_eq!(foo.load(Ordering::SeqCst), 10);
                 }
             }
 
+            doc_comment! {
+                concat!("Subtracts from the given atomic value, returning the previous value.
+
+This operation wraps around on overflow.
+
+`fetch_sub_explicit` takes ", stringify!($atomic_type), " which will be substracted by given value
+in respect to [`Ordering`] argument which describes the memory ordering
+of this operation. All ordering modes are possible. Note that using
+[`Acquire`] makes the store part of this operation [`Relaxed`], and
+using [`Release`] makes the load part [`Relaxed`].
+
+[`Ordering`]: enum.Ordering.html
+[`Relaxed`]: enum.Ordering.html#variant.Relaxed
+[`Release`]: enum.Ordering.html#variant.Release
+[`Acquire`]: enum.Ordering.html#variant.Acquire
+
+# Examples
+
+```
+", $extra_feature, "use std::sync::atomic::{", stringify!($atomic_type), ", Ordering};
+
+let foo = ", stringify!($atomic_type), "::new(20);
+assert_eq!(", stringify!($atomic_type), ".fetch_sub_explicit(foo, Ordering::SeqCst), 20);
+assert_eq!(foo.load(Ordering::SeqCst), 10);
+            ```"),
+                #[inline]
+                #[$stable]
+                #[cfg(target_has_atomic = "cas")]
+                pub fn fetch_sub_explicit(f: *const $atomic_type,
+                                             val: $int_type,
+                                             order: Ordering) -> $int_type {
+                    unsafe { atomic_sub((*f).v.get(), val, order) }
```

Comment:
What's the type of `atomic_sub`? Won't this create a reference again?

Reply:
`*mut T`:

https://github.com/rust-lang/rust/blob/5f36dc23e56b8c53d1718c97f8d531054ef74ccc/src/libcore/sync/atomic.rs#L2176

Reply:
Okay, thanks. So this would indeed fix the problem even under the most strict semantics -- but at the cost of duplicating the entire API surface of the `Atomic*` types.

Reply:
True that. See my comment under the PR description:

> * I am unsure how this can be done without extending the public api of `AtomicUsize`.

edit: basically all `Atomic*` i meant.